media_driver/linux/common/vp/ddi/media_libva_vp.c

```diff
@@ -1142,6 +1143,27 @@ DdiVp_SetProcPipelineParams(
         }
         break;
     }
+
+    switch (pPipelineParam->processing_mode)
+    {
+    case VAProcPowerSavingMode:
+        if (pVpHalSrcSurf->SurfType == SURF_IN_PRIMARY)
+        {
+            pVpHalSrcSurf->ScalingMode       = VPHAL_SCALING_AVS;
+            pVpHalSrcSurf->ScalingPreference = VPHAL_SCALING_PREFER_SFC;
+        }
+        break;
+    case VAProcPerformanceMode:
```

Comment:
I doubt AVS can show better performance.  Probably using VAProcDefaultMode is better here? Any other thought?

Reply:
Generally, AVS sampler has better performance for typical usage(such as scaling, CSC etc.). We have related performance data to prove that.

Reply:
And if the VAAPI user does not set processing mode, we used the default mode(value = 0) which makes it compatible with the previous one.